src/main/java/com/aws/greengrass/tes/CredentialRequestHandler.java

```diff
@@ -115,16 +126,31 @@ public CredentialRequestHandler(final IotCloudHelper cloudHelper, final IotConne
 
         deviceConfiguration.getIotRoleAlias().subscribe((why, newv) -> {
             String iotRoleAlias = Coerce.toString(newv);
-            clearCache();
-            setIotCredentialsPath(iotRoleAlias);
+            if (Utils.isNotEmpty(iotRoleAlias)) {
+                clearCache();
+                setIotCredentialsPath(iotRoleAlias);
+            }
         });
         deviceConfiguration.getThingName().subscribe((why, newv) -> {
-            clearCache();
-            setThingName(Coerce.toString(newv));
+            String v = Coerce.toString(newv);
+            if (Utils.isNotEmpty(v)) {
+                clearCache();
+                setThingName(v);
+            }
         });
         deviceConfiguration.getCertificateFilePath().subscribe((why, newv) -> clearCache());
         deviceConfiguration.getRootCAFilePath().subscribe((why, newv) -> clearCache());
         deviceConfiguration.getPrivateKeyFilePath().subscribe((why, newv) -> clearCache());
+        Topic tesV1Topic = deviceConfiguration.getNucleusConfig().lookup("useTesV1");
+        tesV1Topic.subscribe((why, newv) -> {
+            if (tesV1Topic.getOnce() == null) {
+                setIotCredentialsPath(Coerce.toString(deviceConfiguration.getIotRoleAlias()));
+            } else {
+                if (Coerce.toBoolean(tesV1Topic)) {
+                    this.iotCredentialsPath = "/greengrass/assumeRoleForGroup";
+                }
```

Comment:
If the customer sets `useTesV1` to true and then to false, won't this fail to set the credential path back to V2 TES?

Reply:
Done.